src/main/java/stream/flarebot/flarebot/FlareBotManager.java

```diff
@@ -102,28 +93,10 @@ public void executeCreations() {
     }
 
     private void initGuildSaving() {
-        guilds = new ExpiringMap<>(new ExpiredEvent<String, GuildWrapper>() {
-            @Override
-            public void run(String guildId, GuildWrapper guildWrapper, long expired, long last_retrieved) {
-                //ApiRequester.requestAsync(ApiRoute.UNLOAD, );
-                if ((System.currentTimeMillis() - INACTIVITY_CHECK) > guilds.getLastRetrieved(guildId))
-                    saveGuild(guildId, guildWrapper, last_retrieved);
-                else {
-                    setCancelled(true);
-                    guilds.resetTime(guildId, GUILD_EXPIRE);
-                    saveGuild(guildId, guildWrapper, last_retrieved);
-                }
-            }
-        });
-        new FlareBotTask("Guild Activity Purge") {
-            @Override
-            public void run() {
-                if (!FlareBot.EXITING.get())
-                    guilds.purge();
-                else
-                    cancel();
-            }
-        }.repeat(30_000, 30_000);
+        guilds = CacheBuilder.newBuilder()
+                .expireAfterAccess(GUILD_EXPIRE, TimeUnit.MILLISECONDS)
+                .removalListener((RemovalListener<String, GuildWrapper>) removalNotification -> saveGuild(removalNotification.getKey(), removalNotification.getValue(), -1))
```

Comment:
why is the lastAccessTime -1 here?

Reply:
I personally think we should remove lastAccessTime all together. It serves not real use in the new cache system.

Reply:
It has nothing to do with the cache system, it is for ordering in Cassandra, means guilds that are active get loaded quickly. It is all for making loading as quick as possible.